core/tests/ts-integration/tests/l1.test.ts

```diff
@@ -44,6 +52,31 @@ describe('Tests for L1 behavior', () => {
         errorContract = await deployContract(alice, contracts.errors, []);
     });
 
+    test('Should provide allowance to shared bridge, if base token is not ETH', async () => {
+        const baseTokenAddress = process.env.CONTRACTS_BASE_TOKEN_ADDR!;
+        isETHBasedChain = baseTokenAddress == zksync.utils.ETH_ADDRESS_IN_CONTRACTS;
+        if (!isETHBasedChain) {
+            const baseTokenDetails = testMaster.environment().baseToken;
+            const maxAmount = await alice.getBalanceL1(baseTokenDetails.l1Address);
+            await (await alice.approveERC20(baseTokenDetails.l1Address, maxAmount)).wait();
+        }
+    });
+
+    test('Should calculate l2 base cost, if base token is not ETH', async () => {
+        const gasPrice = await scaledGasPrice(alice);
+        if (!isETHBasedChain) {
+            expectedL2Costs = (
```

Comment:
shouldn't we compare this value with something? or do we want to just calcualte it?

Reply:
it is then used later in the test